src/cache/cache.rs

```diff
@@ -230,23 +230,48 @@ pub fn storage_from_environment(pool: &CpuPool, _handle: &Handle) -> Arc<Storage
     }
 
     if cfg!(feature = "gcs") {
-        if let (Ok(bucket), Ok(key_path)) =
-            (env::var("SCCACHE_GCS_BUCKET"), env::var("SCCACHE_GCS_KEY_PATH"))
+        if let Ok(bucket) = env::var("SCCACHE_GCS_BUCKET")
         {
-            let base_url = match env::var("SSCACHE_GCS_BASE_URL") {
-                Ok(base_url) => base_url,
-                _ => {
-                    let default = "https://www.googleapis.com";
-                    warn!("No SSCACHE_GCS_BASE_URl specified, using default: {}", default);
-                    default.to_owned()
-                }
-            };
-            debug!("Trying GCS({})", base_url);
+            debug!("Trying GCS bucket({})", bucket);
             #[cfg(feature = "gcs")]
             {
+                let base_url = match env::var("SSCACHE_GCS_BASE_URL") {
+                    Ok(base_url) => base_url,
+                    _ => {
+                        let default = "https://www.googleapis.com";
+                        warn!("No SCCACHE_GCS_BASE_URl specified, using default: {}", default);
+                        default.to_owned()
+                    }
+                };
+
+                let cred_path = match env::var("SCCACHE_GCS_KEY_PATH") {
+                    Ok(cred_location) => Some(cred_location),
+                    _ => {
+                        warn!(
+                          "No SCCACHE_GCS_KEY_PATH specified-- no authentication will be used.");
+                        None
+                    }
+                };
+
+                let rw_mode = match env::var("SCCACHE_RW_MODE").as_ref().map(String::as_str) {
+                    Ok("READ_ONLY") => RWMode::ReadOnly,
+                    Ok("READ_WRITE") => RWMode::ReadWrite,
+                    Ok(_) => {
+                        warn!("Invalid SCCACHE_RW_MODE-- defaulting to READ_ONLY.");
```

Comment:
I would like to provide a way for storage backends to error during initialization and have it produce an immediate error on the client. It would be much nicer to get a clear error message in these cases rather than digging through logs to try to figure it out. (Nothing you need to do here, just a note.)